### PR TITLE
Changes for osie to have centralized logging in tinkerbell

### LINKED ISSUE #58: Send the worker logs to the boots syslog server?
https://github.com/tinkerbell/osie/pull/51 dropped support for forwarding the logs to a central place but left the worker without any way to troubleshoot the actions (and since the containers are deleted, we cannot use docker logs).

We should have a way to see the actions/worker logs somewhere, maybe forward them to the boots syslog server?

This is somewhat related to https://github.com/tinkerbell/osie/pull/48, but saving the logs to a local file is not enough because when the worker reboots we loose the logs.

### PR DESCRIPTION
**Pull request by Cbkhare**
Signed-off-by: cbkhare <Chitrabasukhare89@gmail.com>

## Description

Changes to add centralized logging for the Tinkerbell. 

## Why is this needed

- We need a log aggregator for Tinkerbell which can collect log from tink services and tink worker containers. 
- boots PR: https://github.com/tinkerbell/boots/pull/73
- Tink PR: https://github.com/tinkerbell/tink/pull/267
- RFD: https://github.com/tinkerbell/proposals/pull/7

Fixes: #

## How Has This Been Tested?
- [x] Tested with basic hello-world example.
- [x] Verified the testing of logrotation.
- [x] Added below test results. 

**log_aggregation**
```
vagrant@provisioner:/vagrant/deploy$ ls -lrt   /var/log/dockerlfs/
total 20
drwxr-xr-x 2 syslog syslog 4096 Sep  8 12:59 provisioner
-rw-r----- 1 syslog adm    9521 Sep  8 13:04 daemon.log
drwxr-xr-x 2 syslog syslog 4096 Sep  8 13:17 192.168.1.5
vagrant@provisioner:/vagrant/deploy$ ls -lrt   /var/log/dockerlfs/192.168.1.5/
total 12
-rw-r----- 1 syslog adm 2985 Sep  8 13:17 hello_world_0e65404d-5abe-4746-b6a9-bba9be867884.log
-rw-r----- 1 syslog adm 4789 Sep  8 13:17 tink-worker.log
```

**logrotation**:
```
vagrant@provisioner:/vagrant/deploy$ ls -lrt   /var/log/dockerlfs/provisioner/
total 36
-rw-r----- 1 syslog adm  513 Sep  8 13:04 deploy_db_1.log-20200908131599572256.gz
-rw-r----- 1 syslog adm 4099 Sep  8 13:17 deploy_boots_1.log-20200908131599572256.gz
-rw-r----- 1 syslog adm  577 Sep  8 13:17 deploy_nginx_1.log-20200908131599572256.gz
-rw-r----- 1 syslog adm  400 Sep  8 13:17 deploy_hegel_1.log-20200908131599572256.gz
-rw-r----- 1 syslog adm 1450 Sep  8 13:17 deploy_tink-server_1.log-20200908131599572256.gz
-rw-r----- 1 syslog adm 6155 Sep  8 13:37 deploy_registry_1.log-20200908131599572256.gz
-rw-r----- 1 syslog adm    0 Sep  8 13:37 deploy_boots_1.log
-rw-r----- 1 syslog adm    0 Sep  8 13:37 deploy_db_1.log
-rw-r----- 1 syslog adm    0 Sep  8 13:37 deploy_hegel_1.log
-rw-r----- 1 syslog adm    0 Sep  8 13:37 deploy_nginx_1.log
-rw-r----- 1 syslog adm    0 Sep  8 13:37 deploy_tink-server_1.log
-rw-r----- 1 syslog adm  147 Sep  8 13:37 deploy_registry_1.log
vagrant@provisioner:/vagrant/deploy$ ls -lrt   /var/log/dockerlfs/192.168.1.5/
total 8
-rw-r----- 1 syslog adm 571 Sep  8 13:17 hello_world_0e65404d-5abe-4746-b6a9-bba9be867884.log-20200908131599572256.gz
-rw-r----- 1 syslog adm 918 Sep  8 13:17 tink-worker.log-20200908131599572256.gz
-rw-r----- 1 syslog adm   0 Sep  8 13:37 hello_world_0e65404d-5abe-4746-b6a9-bba9be867884.log
-rw-r----- 1 syslog adm   0 Sep  8 13:37 tink-worker.log
vagrant@provisioner:/vagrant/deploy$ ls -lrt   /var/log/dockerlfs/
total 12
-rw-r----- 1 syslog adm    1642 Sep  8 13:04 daemon.log-20200908131599572256.gz
-rw-r----- 1 syslog adm       0 Sep  8 13:37 daemon.log
drwxr-xr-x 2 syslog syslog 4096 Sep  8 13:37 192.168.1.5
drwxr-xr-x 2 syslog syslog 4096 Sep  8 13:37 provisioner

``` 

## How are existing users impacted? What migration steps/scripts do we need?

- Log location has been updated. Documentation will need an update.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)

### CODE REVIEW

File: apps/workflow-helper.sh [hunk 1]
```diff
@@ -11,6 +11,10 @@ registry_password=$(sed -nr 's|.*\bregistry_password=(\S+).*|\1|p' /proc/cmdline
 tinkerbell=$(sed -nr 's|.*\btinkerbell=(\S+).*|\1|p' /proc/cmdline)
 worker_id=$(sed -nr 's|.*\bworker_id=(\S+).*|\1|p' /proc/cmdline)
 id=$(curl --connect-timeout 60 "$tinkerbell:50061/metadata" | jq -r .id)
+log_driver=$(sed -nr 's|.*\blog_driver=(\S+).*|\1|p' /proc/cmdline)
+log_opt_tag=$(sed -nr 's|.*\blog_opt_tag=(\S+).*|\1|p' /proc/cmdline)
+log_opt_server_address=$(sed -nr 's|.*\blog_opt_server_address=(\S+).*|\1|p' /proc/cmdline)
+log_opt_server_address_type=$(sed -nr 's|.*\blog_opt_server_address_type=(\S+).*|\1|p' /proc/cmdline)
```
Comment (mmlb) [inline apps/workflow-helper.sh]: I thought we came to the agreement that we would not add the `address_type` arg and instead do
```
--log-opt "$log_driver-address=$log_opt_server_address
```

Also I thought we were going to go with
```
LOG_DRIVER
LOG_TAG
LOG_SERVER_ADDRESS
LOG_OPT_(all the other options)
```
I'm ok with doing `LOG_DRIVER` and `LOG_OPT` for all the options just wanted to ask.

**Review by mmlb — CHANGES_REQUESTED** [review]

File: apps/workflow-helper.sh (hunk 1 above)
Comment (Cbkhare) [reply apps/workflow-helper.sh]: I have updated the `address-type`. 

For options, I think we concluded to have `OPT` for all the `log-options`.

File: apps/workflow-helper.sh [hunk 2]
```diff
@@ -71,8 +74,14 @@ docker run --privileged -t --name "tink-worker" \
...
 	-e "REGISTRY_PASSWORD=$registry_password" \
+	-e "LOG_DRIVER=$log_driver" \
+	-e "LOG_OPT_SERVER_ADDRESS=$log_opt_server_address" \
+	-e "LOG_OPT_TAG=$log_opt_tag" \
 	-v /worker:/worker \
 	-v /var/run/docker.sock:/var/run/docker.sock \
 	-t \
 	--net host \
+	--log-driver "$log_driver" \
+	--log-opt "$log_driver"-address="$log_opt_server_address" \
```
Comment (mmlb) [inline apps/workflow-helper.sh]: last nit: Please just do one set of `"` for the whole arg.

**Review by mmlb — CHANGES_REQUESTED** [review]

File: apps/workflow-helper.sh (hunk 2 above)
Comment (Cbkhare) [reply apps/workflow-helper.sh]: done

**Pull request closed without merging**